src/api/GraphQuery/queryBuilder.ts

```diff
@@ -208,15 +208,33 @@ export interface BaseStrictPropertyFilter<T> extends BasePropertyFilter {
   operator: Exclude<BasePropertyFilter['operator'], '~'>;
   input: QueryProperty<T>;
 }
+export const BASE_OPERATORS = [
+  '=',
+  '!=',
+  '<',
+  '<=',
+  '>',
+  '>=',
+  '~',
+  'isNull',
+  'isNotNull'
+] as const;
+export type BaseOperators = (typeof BASE_OPERATORS)[number];
+
+export const STRING_OPERATORS = ['=', '!=', 'isNull', 'isNotNull'] as const;
+export type StringOperators = (typeof STRING_OPERATORS)[number];
+
+export const NUMBER_OPERATORS = ['>', '<', '=', '<=', '>=', '!=', 'isNull', 'isNotNull'] as const;
+export type NumberOperators = (typeof NUMBER_OPERATORS)[number];
```

Comment:
We should use the singular form for the type:

```suggestion
export type BaseOperator = (typeof BASE_OPERATORS)[number];

export const STRING_OPERATORS = ['=', '!=', 'isNull', 'isNotNull'] as const;
export type StringOperator = (typeof STRING_OPERATORS)[number];

export const NUMBER_OPERATORS = ['>', '<', '=', '<=', '>=', '!=', 'isNull', 'isNotNull'] as const;
export type NumberOperator = (typeof NUMBER_OPERATORS)[number];
```